controls/views.py

```diff
@@ -921,25 +921,23 @@ def import_component(request):
     return component_library(request)
 
 
-def raise_404_if_not_permitted_to_statement(request, statement, system_permission='view_system'):
-    """Raises a 404 if the user doesn't have the statement system permission"""
-    while True:
-        if request.user.is_superuser or request.user.is_staff:
-            # Allow access if the user has superuser or staff member permissions respectively
-            break
-        else:
-            # Allow access if the user has the statement system permission
-            try:
-                system = System.objects.get(root_element=statement.consumer_element)
-                if request.user.has_perm(system_permission, system):
-                    break
-            except System.DoesNotExist:
-                logger.info(
-                event="update_smt_permission permission_denied",
-                object={"object": "statement", "id": statement.id},
-                user={"id": request.user.id, "username": request.user.username}
-                )
-                raise Http404
+def raise_404_if_statement_permission_not(request, statement, action, system_permission='view_system'):
+    """Raise an HTTP 404 if the user doesn't have the statement system permission"""
+
+    # Allow access if the user has superuser or staff member permissions respectively
+    if not (request.user.is_superuser or request.user.is_staff):
+
+        # Allow access if the user has the statement system permission
+        system = get_object_or_404(System, root_element=statement.consumer_element)
+        raise_404_if_system_permissions_not(request, system, system_permission,
+            lambda permission, subaction, user_message:
+                log_system_object_action(request, system,
```

Comment:
add this to PR

Reply:
@davidpofo Made this independent of #1468